src/mongo/db/dbcommands.cpp

```diff
@@ -367,6 +368,16 @@ class CmdProfile : public Command {
             serverGlobalParams.slowMS = slow.numberInt();
         }
 
+        const BSONElement ratelimitelem = cmdObj["ratelimit"];
+        if (ratelimitelem.isNumber()) {
+            int rateLimit = ratelimitelem.numberInt();
+            if (rateLimit == 0)
```

Comment:
We need checks here, not just an assignment. So I added comments to this part of code and made code more clear (i believe).